### PR TITLE
Allow PublishPorts to be used with [Pod] entry

### PR DESCRIPTION
**Pull request by rhatdan**
Fixed: https://github.com/containers/podman/issues/21035

#### Does this PR introduce a user-facing change?

```release-note
[Pod] quadlets now support PublishPorts.
```

### CODE REVIEW

**Review by ygalblum — APPROVED** [review]
LGTM

File: test/e2e/quadlet/network.pod [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ## assert-podman-pre-args "--network=host"
+## assert-podman-args --publish 127.0.0.1:80:90
```
Comment (ygalblum) [inline test/e2e/quadlet/network.pod:2]:
```suggestion
## assert-podman-pre-args --publish 127.0.0.1:80:90
```

**Review by ygalblum — COMMENTED** [review]
Sorry, missed this in my initial review. This is why the tests are failing

**Review by ygalblum — CHANGES_REQUESTED** [review]
See my comment on the test

**Review by ygalblum — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
@mheon @Luap99 @baude @ashley-cui @umohnani8 @giuseppe PTAL

**Review by ashley-cui — COMMENTED** [review]
/hold
/lgtm

**Pull request merged**